File: src/main/java/com/example/travelator/beansToValue/java/UserPreferences.java
package com.example.travelator.beansToValue.java;

import java.util.Currency;
import java.util.Locale;

/**
 * 동등성과 해시 코드가 가변 프로퍼티 값에 따라 결정
 * -> UserPreferences를 집합에 넣어거나 맵의 키로 사용할 수 없다.
 */
public class UserPreferences {
	private String greeting;
	private Locale locale;
	private Currency currency;
	
	public UserPreferences() {
		this("Hello", Locale.UK, Currency.getInstance(Locale.UK));
	}
	
	public UserPreferences(String greeting, Locale locale, Currency currency) {
		this.greeting = greeting;
		this.locale = locale;
		this.currency = currency;
	}
	
	public String getGreeting() {
		return greeting;
	}
	
	public void setGreeting(String greeting) {
		this.greeting = greeting;
	}
	
	public Locale getLocale() {
		return locale;
	}
	
	public void setLocale(Locale locale) {
		this.locale = locale;
	}
	
	public Currency getCurrency() {
		return currency;
	}
	
	public void setCurrency(Currency currency) {
		this.currency = currency;
	}
}

File: src/main/java/com/example/travelator/beansToValue/kotlin/version1/UserPreferences.kt
package com.example.travelator.beansToValue.kotlin.version1

import java.util.*

/**
 * 코틀린 변환 1차 버전
 */
class UserPreferences
// 컴파일러가 greeting, locale, currency의 디폴트 값들을 서로 조합한(경우의 수) 여러 생성자를 만들어내도록 지정
@JvmOverloads constructor(
    var greeting: String = "Hello",
    var locale: Locale = Locale.UK,
    var currency: Currency = Currency.getInstance(Locale.UK)
)